File: src/main/java/nl/shadowlink/tools/shadowlib/ipl/Item_CULL.java
package nl.shadowlink.tools.shadowlib.ipl;

import nl.shadowlink.tools.io.ReadFunctions;
import nl.shadowlink.tools.io.Vector3D;
import nl.shadowlink.tools.shadowlib.utils.GameType;
import nl.shadowlink.tools.shadowlib.utils.hashing.HashTable;

/**
 * @author Shadow-Link
 */
public class Item_CULL extends IplItem {
    private GameType gameType;
    public Vector3D posLowerLeft;
    public Vector3D posUpperRight;
    public int unk1, unk2, unk3, unk4;
    public long hash;
    public String name;

    Item_CULL(GameType gameType) {
        this.gameType = gameType;
    }

    @Override
    public void read(String line) {
        // Message.displayMsgHigh(line);
    }

    @Override
    public void read(ReadFunctions rf) {
        posLowerLeft = rf.readVector3D();
        posUpperRight = rf.readVector3D();
        unk1 = rf.readInt();
        unk2 = rf.readInt();
        unk3 = rf.readInt();
        unk4 = rf.readInt();
        hash = rf.readInt();
        // display();
    }

    private void display() {
        System.out.println("Position: " + posLowerLeft.x + ", " + posLowerLeft.y + ", " + posLowerLeft.z);
        System.out.println("Rotation: " + posUpperRight.x + ", " + posUpperRight.y + ", " + posUpperRight.z);
        System.out.println("Hash: " + hash);// + " name: " + name);
        System.out.println("Unknown1: " + unk1);
        System.out.println("Unknown2: " + unk2);
        System.out.println("Unknown3: " + unk3);
        System.out.println("Unknown4: " + unk4);
        System.out.println("Name: " + name);
    }

    @Override
    public void read(ReadFunctions rf, HashTable hashTable) {
        posLowerLeft = rf.readVector3D();
        posUpperRight = rf.readVector3D();
        unk1 = rf.readInt();
        unk2 = rf.readInt();
        unk3 = rf.readInt();
        unk4 = rf.readInt();

        long tempHash = rf.readUnsignedInt();
        name = "" + tempHash;
        hash = (int) tempHash;
        name = hashTable.get(tempHash);

        display();
    }

}

File: src/main/java/nl/shadowlink/tools/shadowlib/ipl/Item_CULL.kt
package nl.shadowlink.tools.shadowlib.ipl

import nl.shadowlink.tools.io.ReadFunctions
import nl.shadowlink.tools.io.Vector3D
import nl.shadowlink.tools.shadowlib.utils.GameType
import nl.shadowlink.tools.shadowlib.utils.hashing.HashTable

/**
 * @author Shadow-Link
 */
class Item_CULL internal constructor(private val gameType: GameType) : IplItem() {
    var posLowerLeft: Vector3D? = null
    var posUpperRight: Vector3D? = null
    var unk1 = 0
    var unk2 = 0
    var unk3 = 0
    var unk4 = 0
    var hash: Long = 0
    var name: String? = null
    override fun read(line: String) {
        // Message.displayMsgHigh(line);
    }

    override fun read(rf: ReadFunctions) {
        posLowerLeft = rf.readVector3D()
        posUpperRight = rf.readVector3D()
        unk1 = rf.readInt()
        unk2 = rf.readInt()
        unk3 = rf.readInt()
        unk4 = rf.readInt()
        hash = rf.readInt().toLong()
        // display();
    }

    private fun display() {
        println("Position: " + posLowerLeft!!.x + ", " + posLowerLeft!!.y + ", " + posLowerLeft!!.z)
        println("Rotation: " + posUpperRight!!.x + ", " + posUpperRight!!.y + ", " + posUpperRight!!.z)
        println("Hash: $hash") // + " name: " + name);
        println("Unknown1: $unk1")
        println("Unknown2: $unk2")
        println("Unknown3: $unk3")
        println("Unknown4: $unk4")
        println("Name: $name")
    }

    override fun read(rf: ReadFunctions, hashTable: HashTable) {
        posLowerLeft = rf.readVector3D()
        posUpperRight = rf.readVector3D()
        unk1 = rf.readInt()
        unk2 = rf.readInt()
        unk3 = rf.readInt()
        unk4 = rf.readInt()
        val tempHash = rf.readUnsignedInt()
        name = "" + tempHash
        hash = tempHash.toInt().toLong()
        name = hashTable!![tempHash]
        display()
    }
}